src/Application/Commands/CopyExistingPersonRecordsToMatchbot.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace BigGive\Identity\Application\Commands;
+
+use BigGive\Identity\Application\Auth\Token;
+use BigGive\Identity\Repository\PersonRepository;
+use Doctrine\DBAL\Connection;
+use Monolog\DateTimeImmutable;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\RoutableMessageBus;
+
+/**
+ * Command only needed temporarily, to send copies of all Person records that were last updated before
+ * we started a live sync to matchbot. Once the job is complete this can be deleted.
+ */
+#[AsCommand(name: 'identity:copy-existing-person-records-to-matchbot')]
+class CopyExistingPersonRecordsToMatchbot extends Command
+{
+    /** @psalm-suppress PossiblyUnusedMethod */
+    public function __construct(
+        private Connection $connection,
+        private RoutableMessageBus $bus,
+        private PersonRepository $personRepository,
+        private LoggerInterface $logger
+    ) {
+        parent::__construct();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $personsAlreadySyncedUpTo = new \DateTimeImmutable((string) $this->connection->fetchOne(
+            "SELECT value from `KeyValue` where `Key` = 'PersonRecordsSentToMatchBotTo'"
+        ));
+
+        $batchOfPersons = $this->personRepository->findOldestPersonRecordsRequiringSyncToMatchbot(
+            $personsAlreadySyncedUpTo
+        );
+
+        if ($batchOfPersons === []) {
+            // Sadly I think this condition can never be met - the final batch will have one person not zero, because
```

Comment:
Yep, seems like a reasonable trade off to keep the main tables as simple as possible